drivers/include/hdc2010.h

```diff
@@ -0,0 +1,208 @@
+ *
+ * This allows the user to define custom interrupt logic.
+ */
+ typedef void (*hdc2010_int_cb_t)(void *arg);
+
+/**
+ * @brief   Parameters needed for device initialization
+ */
+typedef struct {
+    i2c_t i2c;                  /**< bus the device is connected to */
+    uint8_t addr;               /**< address on that bus */
+    gpio_t irq_pin;             /**< IRQ pin to check data ready */
+    hdc2010_res_t res;          /**< resolution used for sampling temp and hum */
+    hdc2010_meas_mode_t mode;   /**< sensors to be used */
+    hdc2010_amm_t amm;          /**< Auto Measurement Mode */
+    hdc2010_int_cb_t user_cb;   /**< User defined interrupt callback */
+} hdc2010_params_t;
+
+/**
+ * @brief   Device descriptor for HDC2010 sensors
+ */
+typedef struct {
+    hdc2010_params_t p;     /**< Configuration parameters */
+    mutex_t mutex;          /**< Mutex for this hdc2010 dev */
+} hdc2010_t;
+
+/**
+ * @brief   Initialize the given HDC2010 device
+ *
+ * @param[out] dev          device descriptor of sensor to initialize
+ * @param[in]  params       configuration parameters
+ *
+ * @return                  HDC2010_OK on success
+ * @return                  HDC2010_NOBUS if initialization of I2C bus fails
+ * @return                  HDC2010_NODEV if no HDC2010 device found on bus
+ */
+int hdc2010_init(hdc2010_t *dev, const hdc2010_params_t *params);
+
+/**
+ * @brief   Trigger a new conversion
+ *
+ * After the conversion is triggered, one has to wait
+ * @ref CONFIG_HDC2010_CONVERSION_TIME us until the results can be read using
+ * @ref hdc2010_get_results().
+ *
+ * @param[in]  dev          device descriptor of sensor
+ *
+ * @return                  HDC2010_OK on success
+ * @return                  HDC2010_BUSERR on I2C communication failures
+ */
+int hdc2010_trigger_conversion(const hdc2010_t *dev);
+
+/**
+ * @brief   Read conversion results for temperature and humidity
+ *
+ * @param[in]  dev          device descriptor of sensor
+ * @param[out] temp         temperature [in 100 * degree centigrade]
+ * @param[out] hum          humidity [in 100 * percent relative]
+ *
+ * @return                  HDC2010_OK on success
+ * @return                  HDC2010_BUSERR on I2C communication failures
+ */
+int hdc2010_get_results(const hdc2010_t *dev, int16_t *temp, int16_t *hum);
+
+/**
+ * @brief   Convenience function for reading temperature and humidity
+ *
+ * This function will trigger a new conversion, wait for the conversion to be
+ * finished and the get the results from the device.
+ *
+ * @param[in]  dev          device descriptor of sensor
+ * @param[out] temp         temperature [in 100 * degree centigrade]
+ * @param[out] hum          humidity [in 100 * percent relative]
+ *
+ * @return                  HDC2010_OK on success
+ * @return                  HDC2010_BUSERR on I2C communication failures
+ */
+int hdc2010_read(hdc2010_t *dev, int16_t *temp, int16_t *hum);
```

Comment:
hum can't be negative, right?

Reply:
indeed, it should be `uint16_t`

Reply:
although just a note: `phydat_t.val` is an `int16_t[3]`... hence there's gonna be a casting going on in saul if we change here to `uint16_t`. shouldn't be a problem since the value is between 0-10000, i.e. no negative value... just saying that imo it's not that nice that one is force to do an explicit cast there..